### PR TITLE
Refactored out into the Shared/Tests solution directory

### LINKED ISSUE #209: For powershell scripts, switch default command from "powershell" to "pwsh"
Primarily this is probably going to be a chapter 10 issue

### PR DESCRIPTION
**Pull request by MarkMichaelis**
Refactored out a file into the Shared/Tests solution directory that may be used to execute PowerShell scripts.  This is used to execute the HelloWorldInC#.ps1 file.

Fixes #209

### CODE REVIEW

File: src/Chapter10.Tests/Listing10.13.MakingTypesAvailableOutsideAnAssembly.Tests.cs [hunk 1]
```diff
@@ -14,26 +13,15 @@ public class TypeTests
         [TestMethod]
         public void MakingTypesAvailableExternallyPS1_ExitCodeIs0()
         {
-            if (PowerShellTestsUtilities.PowerShellNotInstalled()) Assert.Inconclusive("Powershell not installed");
+            if (PowerShellTestUtilities.PowerShellNotInstalled) Assert.Inconclusive("Powershell not installed");
             //EssentialCSharp\\src\\Chapter10.Tests\\bin\\Debug\\netcoreapp3.0
             string ps1Path = Path.GetFullPath("../../../../Chapter10/Listing10.13.MakingTypesAvailableExternally.ps1", Environment.CurrentDirectory);
```
Comment (BenjaminMichaelis) [inline src/Chapter10.Tests/Listing10.13.MakingTypesAvailableOutsideAnAssembly.Tests.cs:18]: You can use Intellitect.Multitool.RepositoryPaths.GetDefaultRepoRoot() to get the root of the repo

File: src/Shared/Tests/PowerShellTestUtilities.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    {
+        public static bool WindowsEnvironment()
+        {
+            return RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+        }
+
+
+        public static Lazy<string?> _PowerShellCommand = new Lazy<string?>(() =>
+        {
+            string? powershellCommand = WindowsEnvironment() ? "powershell" : null;
```
Comment (Keboo) [inline src/Shared/Tests/PowerShellTestUtilities.cs]: Is there any reason not to use powershell core on all platforms?

File: src/Shared/Tests/PowerShellTestUtilities.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        public static int RunPowerShellScript(string scriptPath, string arguments, out string psOutput)
+        {
+            if(PowerShellCommand is null)
+            {
+                throw new InvalidOperationException("PowerShell is not installed");
+            }
+
+            if(!System.IO.File.Exists(scriptPath))
+            {
+                throw new InvalidOperationException($"scriptPath, '{scriptPath}', not found.");
```
Comment (Keboo) [inline src/Shared/Tests/PowerShellTestUtilities.cs:52]: nit: This should be an `ArgumentException`

File: src/Shared/Tests/PowerShellTestUtilities.cs [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+            powerShell.StartInfo.FileName = PowerShellCommand;
+            powerShell.StartInfo.Arguments = $"-noprofile -command \"{scriptPath} {arguments}\"";
+            powerShell.Start();
+            psOutput = powerShell.StandardOutput.ReadToEnd();
+            powerShell.WaitForExit();
+
+            return powerShell.ExitCode;
+        }
+
+        public static string? PowerShellCommand => _PowerShellCommand!.Value;
```
Comment (Keboo) [inline src/Shared/Tests/PowerShellTestUtilities.cs:66]: Is the null forgiveness operator needed here? The field is non-nullable.
```suggestion
        public static string? PowerShellCommand => _PowerShellCommand.Value;
```

File: src/Chapter10.Tests/Listing10.23.RegisteringAFinalizerWithProcessExit.Tests.cs [hunk 5]
```diff
@@ -2,62 +2,30 @@
...
 using System.Text.RegularExpressions;
 
 namespace AddisonWesley.Michaelis.EssentialCSharp.Chapter10.Listing10_23.Tests
 {
     [TestClass]
-
     public class DisposeTests
     {
-
         static string Ps1Path { get; } = Path.GetFullPath("../../../../Chapter10/Listing10.23.RegisteringAFinalizerWithProcessExit.ps1", Environment.CurrentDirectory);
```
Comment (BenjaminMichaelis) [inline src/Chapter10.Tests/Listing10.23.RegisteringAFinalizerWithProcessExit.Tests.cs]:
```suggestion
        static string Ps1Path { get; } = Path.GetFullPath(Path.Combine("..,..,..,..,Chapter10,Listing10.23.RegisteringAFinalizerWithProcessExit.ps1"), Environment.CurrentDirectory);
```
Shouldn't this use commas to make it cross-plat compatible?

**Comment by MarkMichaelis** [comment]
This is no longer active.

**Pull request closed without merging**